src/extension/src/scripts/generate-test.ts

```diff
@@ -0,0 +1,129 @@
+import { CoreTemplateStudio } from "../coreTemplateStudio";
+import { CONSTANTS } from "../CONSTANTS";
+
+let instance: CoreTemplateStudio;
+let backend: string[] = [];
+let frontend: string[] = [];
+let pagesObj: any[] = [];
+const projType = "FullStackWebApp";
+let syncAttemptNum = 0;
+
+function delay(time: number) {
+  return new Promise(function(resolve) {
+    setTimeout(resolve, time);
+  });
+}
+
+let attemptSync: any = (
+  instanceObj: CoreTemplateStudio,
+  syncAttemptNum: number
+) => {
+  if (syncAttemptNum < CONSTANTS.API.MAX_SYNC_REQUEST_ATTEMPTS) {
+    return instanceObj
+      .sync({
+        port: instance.getPort(),
+        payload: { path: CONSTANTS.API.DEVELOPMENT_PATH_TO_TEMPLATES },
+        liveMessageHandler: value => {
+          value;
+        }
+      })
+      .then(() => {
+        instance
+          .getFrameworks(projType)
+          .then(frameworks => {
+            frameworks.forEach(
+              (obj: { tags: { type: string }; name: string }) => {
+                if (obj.tags.type == "frontend") {
+                  frontend.push(obj.name);
+                } else if (obj.tags.type == "backend") {
+                  backend.push(obj.name);
+                }
+              }
+            );
+
+            function getPagesObj(
+              instance: CoreTemplateStudio,
+              projType: string,
+              frontend: any,
+              backend: any
+            ) {
+              instance
+                .getPages(projType, frontend, backend)
+                .then(pages => {
+                  pagesObj = [];
+                  pages.forEach((page: { name: any; templateId: any }) => {
+                    pagesObj.push({
+                      name: page.name,
+                      identity: page.templateId
+                    });
+                  });
+                })
+                .catch(err => {
+                  console.log(err);
+                });
+              return pagesObj;
```

Comment:
I think this logic should work, but you're mixing sync and async code here. When you return `pagesObj` here it is undefined as the promise above has not been resolved. You can return the above promise instead and then wait for it in `generateProj`. You can verify this is an issue by looking at one of the generated projects; it has no pages.